### PR TITLE
Modified the code of AzureStorageBlobDataCopyTool (3) ( Added the check for input length.etc ) 

### PR DESCRIPTION
**Pull request by daisukenishino**
1) Added input length check error message in resource file
2) To check for input length of Reserved area
3) Handled while uploading and downloading document

### CODE REVIEW

File: programs/AzureStorageBlobDataCopyTool/WebSite/CopyTool/User/UserScreen_ProvidedVHDList.aspx.cs [hunk 1]
```diff
@@ -180,12 +181,26 @@ protected void UOC_btnCopyStart_Click(FxEventArgs fxEventArgs)
...
-            String vhdName = "VHDName : " + txtDestFileName.Attributes["VHDName"].ToString() + Environment.NewLine + "File Name : " + txtDestFileName.Text;
-            copyInfoParmeterValue.SourceFileName = txtDestFileName.Attributes["VhdFileNameSource"].ToString();
-            copyInfoParmeterValue.DestinationFileName = txtDestFileName.Text;
+            copyInfoParmeterValue.DestinationFileName = txtDestFileName.Text.Trim();
+
+            if (!string.IsNullOrEmpty(copyInfoParmeterValue.DestinationFileName))
+            {
+                String vhdName = "VHDName : " + txtDestFileName.Attributes["VHDName"].ToString() + Environment.NewLine + "File Name : " + copyInfoParmeterValue.DestinationFileName;
+
+                if (vhdName.Length >= 100)
```
Comment (daisukenishino) [inline programs/AzureStorageBlobDataCopyTool/WebSite/CopyTool/User/UserScreen_ProvidedVHDList.aspx.cs]: nochi-san pointed out the check method problem.
- Please implement the check processing of up to 150 characters or less.  
txtDestFileName.Attributes ["VHDName"]. ToString ()
- Please implement the check processing of up to 256 characters or less.  
copyInfoParmeterValue.DestinationFileName

Please extended reserve area corresponding to the above-mentioned.

Comment (SandeepNayak077) [inline programs/AzureStorageBlobDataCopyTool/WebSite/CopyTool/User/UserScreen_ProvidedVHDList.aspx.cs]: Dear Nishino-san,

According to the above comment, the following 3 changes are required in the database 
1) vhdheader (Table)
- vhdname nvarchar(100) --> vhdname nvarchar(150)

2) vhdfile (Table)
- vhdfilename nvarchar(100) --> vhdfilename nvarchar(256)

3) AsyncProcessingServiceStatusManagementTable (Table)
- ReservedArea nvarchar(200) --> ReservedArea nvarchar(10 + 150 + 12 + 256) --> ReservedArea nvarchar(428)

please confirm.

**Comment by SandeepNayak077** [comment]
Dear Nishino-san,

According to the above comment, the following 3 changes are required in the database 
1) vhdheader (Table)
- vhdname nvarchar(100) --> vhdname nvarchar(150)

2) vhdfile (Table)
- vhdfilename nvarchar(100) --> vhdfilename nvarchar(256)

3) AsyncProcessingServiceStatusManagementTable (Table)
- ReservedArea nvarchar(200) --> ReservedArea nvarchar(10 + 150 + 12 + 256) --> ReservedArea nvarchar(428)

please confirm.

File: programs/AzureStorageBlobDataCopyTool/WebSite/CopyTool/User/UserScreen_ProvidedVHDList.aspx.cs (hunk 1 above)
Comment (daisukenishino) [inline programs/AzureStorageBlobDataCopyTool/WebSite/CopyTool/User/UserScreen_ProvidedVHDList.aspx.cs]: - There is no problem with the changes.
- Please set as follows with a margin.  
ReservedArea nvarchar (512)

**Comment by daisukenishino** [comment]
You are using the SQL Server 2008 R2.
But we had to request the use of the Express edition.

**Comment by daisukenishino** [comment]
Internationalization of AzureStorageBlobDataCopyTool correspondence has not been completed.
It is necessary to change the UI culture by Accept-Language attribute of the HTTP request header.
- Setting the Culture and UI Culture for Web Forms Globalization  
https://msdn.microsoft.com/library/bz9tc508(v=vs.71).aspx

And Suggest the implementation location.
- UICulture not accepted in Master pages  The ASP.NET Forums  
http://forums.asp.net/t/1008181.aspx?UICulture+not+accepted+in+Master+pages

**Pull request merged**